Sources/EdgeNetworkHandlers/EdgeHitProcessor.swift

```diff
@@ -80,69 +78,112 @@ class EdgeHitProcessor: HitProcessing {
-                                             requestType: EdgeRequestType.interact,
-                                             requestProperties: requestProperties,
-                                             locationHint: locationHint)
-            let edgeHit = ExperienceEventsEdgeHit(endpoint: endpoint, configId: configId, request: requestPayload)
-            // NOTE: the order of these events needs to be maintained as they were sent in the network request
-            // otherwise the response callback cannot be matched
-            networkResponseHandler.addWaitingEvents(requestId: edgeHit.requestId,
-                                                    batchedEvents: listOfEvents)
-            sendHit(entityId: entity.uniqueIdentifier, edgeHit: edgeHit, headers: getRequestHeaders(event), completion: completion)
-        } else if event.isUpdateConsentEvent {
-            guard let eventData = event.data, !eventData.isEmpty else {
-                Log.debug(label: EdgeConstants.LOG_TAG, "\(SELF_TAG) - Failed to process Consent event, data was nil or empty")
-                completion(true)
-                return
-            }
+        let eventConfig = getEventConfig(from: event)
+        let datastreamIdOverride = eventConfig?[EdgeConstants.EventDataKeys.Config.DATASTREAM_ID_OVERRIDE] as? String ?? ""
 
-            // Build and send the consent network request to Experience Edge
-            guard let consentPayload = requestBuilder.getConsentPayload(event) else {
-                Log.debug(label: EdgeConstants.LOG_TAG,
-                          "\(SELF_TAG) - Failed to build the consent payload, dropping event '\(event.id.uuidString)'.")
-                completion(true)
-                return
-            }
+        // check and udpate datastream id if datastream id override is set
+        if !datastreamIdOverride.isEmpty {
+            datastreamId = datastreamIdOverride
+        }
 
-            let endpoint = buildEdgeEndpoint(config: edgeConfig,
-                                             requestType: EdgeRequestType.consent,
-                                             requestProperties: nil,
-                                             locationHint: locationHint)
-            let edgeHit = ConsentEdgeHit(endpoint: endpoint, configId: configId, consents: consentPayload)
-            networkResponseHandler.addWaitingEvent(requestId: edgeHit.requestId, event: event)
-            sendHit(entityId: entity.uniqueIdentifier, edgeHit: edgeHit, headers: getRequestHeaders(event), completion: completion)
-        } else if event.isResetIdentitiesEvent {
-            // reset stored payloads as part of processing the reset hit
-            let storeResponsePayloadManager = StoreResponsePayloadManager(EdgeConstants.DataStoreKeys.STORE_NAME)
-            storeResponsePayloadManager.deleteAllStorePayloads()
+        let edgeHit = ExperienceEventsEdgeHit(endpoint: endpoint, configId: datastreamId, request: requestPayload)
```

Comment:
- it would good to rename the param name from configId -> datastreamId to reduce confusion (we can make an exception for the edgeConfigId variable since it matches the default config key name).
- you can simplify few lines above with something like: 
ExperienceEventsEdgeHit(endpoint: endpoint, configId: datastreamIdOverride ?? edgeConfigid, request: requestPayload)

Reply:
Should we not check for empty string as datastreamIdOverride then?

Reply:
we should, something like  datastreamIdOverride && !datastreamIdOverride.isEmpty ? datastreamIdOverride : edgeConfigId

Reply:
Swift doesn't have ternary operator. I would rather keep it clear like before.

Reply:
@emdobrin @kevinlind 
I have updated the logic and have still kept a variable as  we need to do 2 things when datastream ID override value is found. 1 is update the datastream ID with the override ID value and other is to send the original datastream ID. Let me know your thoughts on this.